src/System.Private.CoreLib/shared/System/Reflection/SignatureType.cs

```diff
@@ -49,40 +50,40 @@ public sealed override Type MakeArrayType(int rank)
         public sealed override Type MakeGenericType(params Type[] typeArguments) => throw new NotSupportedException(SR.NotSupported_SignatureType); // There is no SignatureType for type definition types so it would never be legal to call this.
 
         // Dissectors
-        public sealed override Type GetElementType() => ElementType;
+        public sealed override Type? GetElementType() => ElementType;
         public abstract override int GetArrayRank();
         public abstract override Type GetGenericTypeDefinition();
         public abstract override Type[] GenericTypeArguments { get; }
         public abstract override Type[] GetGenericArguments();
         public abstract override int GenericParameterPosition { get; }
-        internal abstract SignatureType ElementType { get; }
+        internal abstract SignatureType? ElementType { get; }
 
         // Identity
 #if DEBUG
-        public sealed override bool Equals(object o) => base.Equals(o);
-        public sealed override bool Equals(Type o) => base.Equals(o);
+        public sealed override bool Equals(object? o) => base.Equals(o);
+        public sealed override bool Equals(Type? o) => base.Equals(o);
         public sealed override int GetHashCode() => base.GetHashCode();
 #endif
         public sealed override Type UnderlyingSystemType => this;  // Equals(Type) depends on this.
     
         // Naming and diagnostics
         public abstract override string Name { get; }
-        public abstract override string Namespace { get; }
-        public sealed override string FullName => null;
-        public sealed override string AssemblyQualifiedName => null;
+        public abstract override string? Namespace { get; }
+        public sealed override string? FullName => null;
+        public sealed override string? AssemblyQualifiedName => null;
         public abstract override string ToString();
 
         // Not supported on Signature Types
         public sealed override Assembly Assembly => throw new NotSupportedException(SR.NotSupported_SignatureType);
         public sealed override Module Module => throw new NotSupportedException(SR.NotSupported_SignatureType);
-        public sealed override Type ReflectedType => throw new NotSupportedException(SR.NotSupported_SignatureType);
-        public sealed override Type BaseType => throw new NotSupportedException(SR.NotSupported_SignatureType);
+        public sealed override Type? ReflectedType => throw new NotSupportedException(SR.NotSupported_SignatureType);
```

Comment:
Why are all these return types nullable if the just `throw new NotSupportedException`? and they are sealed, so its derived types can't override these properties? 

cc: @stephentoub I htink all these should return non-nullable, no?

Reply:
For this derived type, it doesn't really matter, as it's internal and we're not going to call them directly.

I'm more interested in the base virtuals. Do any of our implementations return null from any of these?

Reply:
> I'm more interested in the base virtuals. Do any of our implementations return null from these?

Base virtuals returning null https://github.com/dotnet/coreclr/pull/23489/files#diff-fa74510adac94496994a1ba8b278e5d3R33
If any of you prefer make these overrides non nullable please let me know

Reply:
Base ones can return null, but these ones will just throw, I think it makes sense to make them non-nullable since they're overriding and constraining the virtual members.